packages/m2m-gateway/src/api/attributeApiConverter.ts

```diff
@@ -10,30 +10,32 @@ import {
 } from "../utils/validators/attributeValidators.js";
 import { attributeNotFound } from "../model/errors.js";
 
-export function toM2MGatewayApiCertifiedAttribute({
-  attribute,
-  logger,
-  mapThrownErrorsToNotFound = false,
-}: {
-  attribute: attributeRegistryApi.Attribute;
-  logger: Logger;
-  mapThrownErrorsToNotFound?: boolean;
-}): m2mGatewayApi.CertifiedAttribute {
+function convertAttribute<T>(
+  attribute: attributeRegistryApi.Attribute,
+  attributeKind: attributeRegistryApi.AttributeKind,
+  logger: Logger,
+  mapThrownErrorsToNotFound = false
+): T {
   try {
-    assertAttributeKindIs(
-      attribute,
-      attributeRegistryApi.AttributeKind.Values.CERTIFIED
-    );
-    assertAttributeOriginAndCodeAreDefined(attribute);
+    assertAttributeKindIs(attribute, attributeKind);
 
-    return {
+    const baseFields = {
       id: attribute.id,
-      code: attribute.code,
       description: attribute.description,
-      origin: attribute.origin,
       name: attribute.name,
       createdAt: attribute.creationTime,
     };
+
+    if (attributeKind === attributeRegistryApi.AttributeKind.Values.CERTIFIED) {
+      assertAttributeOriginAndCodeAreDefined(attribute);
+      return {
+        ...baseFields,
+        code: attribute.code,
+        origin: attribute.origin,
+      } as T;
+    }
+
+    return baseFields as T;
```

Comment:
I like the helper function!
Maybe here I would use an exhaustive ts-pattern match instead of an if, so that in case in the future we add a new attribute kind, we are forced to add a case here